### PR TITLE
Added Pypi action

### PR DESCRIPTION
**Pull request by confoundry**

### CODE REVIEW

File: .github/workflows/pypi-build.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+
+    - uses: actions/cache@v3
```
Comment (agrinh) [inline .github/workflows/pypi-build.yml:21]: Shall we drop the cache for the publish workflow?

File: .github/workflows/pypi-build.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+on:
```
Comment (agrinh) [inline .github/workflows/pypi-build.yml:1]: Can we name the workflow `pypi-publish`?

File: .github/workflows/pypi-build.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    - name: Upload build artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: Build artifacts
+        path: ./dist
+        retention-days: 90
+
+    - name: Publish package distributions to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
```
Comment (agrinh) [inline .github/workflows/pypi-build.yml:44]: \+`\n`

**Review by agrinh — CHANGES_REQUESTED** [review]

File: .github/workflows/pypi-build.yml (hunk 2 above)
Comment (confoundry) [reply .github/workflows/pypi-build.yml:1]: yep

File: .github/workflows/pypi-build.yml (hunk 1 above)
Comment (confoundry) [reply .github/workflows/pypi-build.yml:21]: sure

File: .github/workflows/pypi-build.yml (hunk 3 above)
Comment (confoundry) [reply .github/workflows/pypi-build.yml:44]: if it makes you happy

File: .github/workflows/pypi-publish.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+on:
+  push:
+    branches: [ "main" ]
+    paths: ["**.py"]  # only push new packages when python files have changed
```
Comment (agrinh) [inline .github/workflows/pypi-publish.yml:4]: We should publish on any changes in `src/causica` (including config files e.g.), changes to dependencies and any change to a markdown file that is included in the distributed files (e.g. the README)

Comment (confoundry) [reply .github/workflows/pypi-publish.yml:4]: used path-ignore to achieve something similar

**Review by agrinh — APPROVED** [review]

File: .github/workflows/pypi-build.yml (hunk 3 above)
Comment (agrinh) [reply .github/workflows/pypi-build.yml]: GitHub and I rejoice: 
image

**Pull request merged**